raiden/waiting.py

```diff
@@ -517,9 +520,9 @@ def wait_for_received_transfer_result(
                 if raiden.get_block_number() <= transfer.lock.expiration:
                     result = TransferWaitResult.SECRET_REGISTERED_ONCHAIN
                     break
-
-        log.debug("wait_for_transfer_result", **log_details)
-        gevent.sleep(retry_timeout)
+        else:
```

Comment:
The usage of `for ... else` works fine, but wouldn't it be clearer to `return result` above instead of using the break?

Reply:
I think I cannot use return here, because the `break` just jumps out of the inner loop. But the second inner loop might overwrite the value of `result` then.